stripe cleanup: Remove the option from create and add-brick cmds

Sheetalpamecha opened the pull request:
This patch aims to remove the code for stripe option instead
of keeping a default values of stripe/stripe-count variables and
setting and getting dict options and similar redundant operations.
Also removing tests for stripe volumes that have been already
marked bad.

Updates: #1000
Change-Id: Ic2b3cabd671f0c8dc0521384b164c3078f7ca7c6
Signed-off-by: Sheetal Pamecha <spamecha@redhat.com>

Sheetalpamecha committed:
stripe cleanup: Remove the option from create and add-brick cmds

This patch aims to remove the code for stripe option instead
of keeping a default values of stripe/stripe-count variables and
setting and getting dict options and similar redundant operations.
Also removing tests for stripe volumes that have been already
marked bad.

Updates: #1000
Change-Id: Ic2b3cabd671f0c8dc0521384b164c3078f7ca7c6
Signed-off-by: Sheetal Pamecha <spamecha@redhat.com>

Sheetalpamecha commented:
/run regression

amarts commented on xlators/mgmt/glusterd/src/glusterd-handler.c:
```diff
@@ -5841,8 +5833,6 @@ glusterd_get_state(rpcsvc_request_t *req, dict_t *dict)
         fprintf(fp, "Volume%d.snap_count: %" PRIu64 "\n", count,
                 volinfo->snap_count);
-        fprintf(fp, "Volume%d.stripe_count: %d\n", count,
-                volinfo->stripe_count);
```
this removal may result in checksum mismatch during update... can you check if we need to write it regardless.

@mohit84 please confirm.

amarts reviewed (commented):
LGTM.

Sheetalpamecha committed:
Fix regression error

tests/000-flaky/basic_changelog_changelog-snapshot.t
was failing due to 0 return value

Change-Id: I8ea0443669c63768760526db5aa1f205978e1dbb
Signed-off-by: Sheetal Pamecha <spamecha@redhat.com>

Sheetalpamecha commented:
/run regression

Sheetalpamecha commented:
/run regression

Sheetalpamecha commented:
Hi @mohit84, Review approval please if you don't see any other issue with this patch.
As discussed in gchat, Should we move on with constant value for stripe as it will never change and serve the purpose for just comparison?

mohit84 commented:
> Hi @mohit84, Review approval please if you don't see any other issue with this patch.
> As discussed in gchat, Should we move on with constant value for stripe as it will never change and serve the purpose for just comparison?

Yes sheetal please do it

Sheetalpamecha committed:
add constant stripe_count value for upgrade scenerios

Change-Id: I49f3da4f106c55f9da20d0b0a299275a19daf4ba

Sheetalpamecha commented:
/run regression

Sheetalpamecha commented:
/recheck smoke

Sheetalpamecha committed:
Fix clang-format warning

Change-Id: I83bae85d10c8c5b3c66f56c9f8de1ec81d0bbc95

Sheetalpamecha commented:
/run regression

Sheetalpamecha commented:
@mohit84  and @amarts please review the patch. Required changes have been done after last discussion.
